### PR TITLE
Add (skopeo generate-sigstore-key)

### PR DESCRIPTION
**Pull request by mtrmac**
Depends on unmerged https://github.com/containers/image/pull/1810 , missing documentation - see FIXMEs

@vrothberg PTAL.

### CODE REVIEW

File: cmd/skopeo/generate_sigstore_key.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+type generateSigstoreKeyOptions struct {
+	outputPrefix   string
+	passphraseFile string
+}
+
+func generateSigstoreKeyCmd() *cobra.Command {
+	var opts generateSigstoreKeyOptions
+	cmd := &cobra.Command{
+		Use:     "generate-sigstore-key --output-prefix PREFIX",
```
Comment (mtrmac) [inline cmd/skopeo/generate_sigstore_key.go:23]: I’m not sure about the `--output-prefix` approach — an alternative is to have `--output-public` and `--output--private`, which is more explicit, but also much more typing.

File: cmd/skopeo/generate_sigstore_key.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+	}
+}
+
+func (opts *generateSigstoreKeyOptions) run(args []string, stdout io.Writer) error {
+	if len(args) != 0 || opts.outputPrefix == "" {
+		return errors.New("Usage: generate-sigstore-key --output-prefix PREFIX")
+	}
+
+	pubKeyPath := opts.outputPrefix + ".pub"
+	privateKeyPath := opts.outputPrefix + ".private"
```
Comment (mtrmac) [inline cmd/skopeo/generate_sigstore_key.go:54]: `.private` or just `.key`?

`.key` is somewhat traditional, OTOH users publish private keys often enough that making that harder seems worthwhile.

File: cmd/skopeo/generate_sigstore_key.go (hunk 1 above)
Comment (vrothberg) [reply cmd/skopeo/generate_sigstore_key.go:23]: I like the idea :+1: Naming is always hard and I do not have another idea.

File: cmd/skopeo/generate_sigstore_key.go (hunk 2 above)
Comment (vrothberg) [reply cmd/skopeo/generate_sigstore_key.go:54]: :+1: on `.private`.  I am very likely such a user that would do it on low caffeine.

**Review by vrothberg — APPROVED** [review]
LGTM (minus docs)

**Comment by mtrmac** [comment]
Now vendors a merged c/image commit, and includes a man page.

Ready for review and possible merging.

File: docs/skopeo-generate-sigstore-key.1.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+Generates a public/private key pair suitable for creating sigstore image signatures.
+The private key is encrypted with a passphrase;
+if one is not provided using an option, this command prompts for it interactively.
+
+The private key is written to _prefix_**.private** .
+The private key is written to _prefix_**.pub** .
+
+## OPTIONS
+
+**--output=prefix** _prefix_
```
Comment (vrothberg) [inline docs/skopeo-generate-sigstore-key.1.md:20]:
```suggestion
**--output-prefix=prefix** _prefix_
```

**Review by vrothberg — COMMENTED** [review]
besides the nit LGTM

File: docs/skopeo-generate-sigstore-key.1.md (hunk 3 above)
Comment (mtrmac) [reply docs/skopeo-generate-sigstore-key.1.md]: Thank you! Fixed I think, not quite as suggested.

**Review by vrothberg — COMMENTED** [review]
LGTM
But you may need a last wrestle with CI:
> hack/man-page-checker
WARNING: skopeo generate-sigstore-key: man page shows '[*options*]', help does not show [command options]
make: *** [Makefile:245: validate-docs] Error 1

**Comment by mtrmac** [comment]
Sorry about that, the script doesn’t run on macOS. It is passing now.

**Pull request merged**